test/quantization/core/experimental/test_nonuniform_observer.py

```diff
@@ -2,22 +2,116 @@
 
 from torch.ao.quantization.experimental.observer import APoTObserver
 import unittest
-import torch
 
 class TestNonUniformObserver(unittest.TestCase):
     def test_calculate_qparams(self):
-        t = torch.Tensor()
-        obs = APoTObserver(t, t, t, 0, 0)
+        """
+        Test case 1
+        Test that error is thrown when k == 0
+        """
+        obs1 = APoTObserver()
 
-        with self.assertRaises(NotImplementedError):
-            obs.calculate_qparams()
+        with self.assertRaises(AssertionError):
+            obs1_result = obs1.calculate_qparams(signed=False)
 
-    def test_override_calculate_qparams(self):
-        t = torch.Tensor()
-        obs = APoTObserver(t, t, t, 0, 0)
+        """
+        Test case 2
+        APoT paper example: https://arxiv.org/pdf/1909.13144.pdf
+        Assume hardcoded parameters:
+        * b = 4 (total number of bits across all terms)
+        * k = 2 (base bitwidth, i.e. bitwidth of every term)
+        * n = 2 (number of additive terms)
+        * note: b = k * n
+        """
+        obs2 = APoTObserver(max_val=1.0, b=4, k=2)
+        obs2_result = obs2.calculate_qparams(signed=False)
 
-        with self.assertRaises(NotImplementedError):
-            obs._calculate_qparams()
+        self.assertEqual(obs2_result[0], (2 / 3))
+
+        quantlevels_size_test2 = int(len(obs2_result[1]))
+        self.assertEqual(quantlevels_size_test2, 16)
+
+        levelindices_size_test2 = int(len(obs2_result[2]))
+        self.assertEqual(levelindices_size_test2, 16)
+
+        unique_elts = True
+        level_indices2_test_list = obs2_result[2].tolist()
+        for i in range(16):
+            if level_indices2_test_list.count(i) != 1:
+                unique_elts = False
+
+        self.assertTrue(unique_elts)
+
+        """
+        Test case 3
+        Assume hardcoded parameters:
+        * b = 6 (total number of bits across all terms)
+        * k = 2 (base bitwidth, i.e. bitwidth of every term)
+        * n = 3 (number of additive terms)
+        """
+        obs3 = APoTObserver(max_val=1.0, b=6, k=2)
+
+        obs3_result = obs3.calculate_qparams(signed=False)
```

Comment:
wondering if it would be better to return a dict instead of a list here?

Reply:
I think returning a tuple is best since we want to return gamma, quantization levels, and level indices which have different types (gamma is a float, the others are tensors).